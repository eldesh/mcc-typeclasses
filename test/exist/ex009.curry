-- Simple context for an existentially quantified variable

data Showable = forall a . Show a => Showable a

instance Show Showable where
  showsPrec p (Showable x) = showsPrec p x

main = mapM_ print [Showable "abc", Showable (Just 123), Showable ((),False,1.0)]
