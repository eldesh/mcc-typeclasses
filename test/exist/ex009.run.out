"abc"
Just 123
((),False,1.0)
